src/Common/Auth/OpenIDConnect/JWT/JsonWebKeySet.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/**
+ * JsonWebKeySet represents a signing key used for JWT signature verification.  It can take a JSON Web Key Set (JWKS)
+ * as a string or as a URI.  If a URI is provided it will retrieve the JWKS and store them internally.
+ *
+ * @package openemr
+ * @link      http://www.open-emr.org
+ * @author    Stephen Nielson <stephen@nielson.org>
+ * @copyright Copyright (c) 2021 Stephen Nielson <stephen@nielson.org>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Common\Auth\OpenIDConnect\JWT;
+
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Psr7\Request;
+use Lcobucci\JWT\Signer\Key;
+use OpenEMR\Common\Logging\SystemLogger;
+use Psr\Http\Client\ClientInterface;
+use Psr\Log\LoggerInterface;
+
+class JsonWebKeySet extends Key
+{
+    /**
+     * @var ClientInterface
+     */
+    private $httpClient;
+
+    private $jwks = [];
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    public function __construct(ClientInterface $httpClient, $jwks_uri = null, $jwks = null)
+    {
+        $this->logger = new SystemLogger();
+        $this->setHttpClient($httpClient);
+
+        $passphrase = '';
```

Comment:
just wondering if passphrase is gonna be used here?

Reply:
Yep. The original Key class takes in a passphrase.  The JWKS never uses the passphrase.